### PR TITLE
Fix for crash in waiting for launchd service start

### PR DESCRIPTION
**Pull request by gabriel**
https://gist.github.com/gabriel/2b090dc0bfce26d008f783b8c9d0c26d

### CODE REVIEW

**Comment by gabriel** [comment]
@jzila @cjb

**Comment by gabriel** [comment]
launchd.LoadStatus can return nil for the status, if for some reason launchd didn't register that the service was loaded... this might happen if launchd is under load or a race possibly?

This fix adds a wait check of up to 5 seconds.

I'm wondering if I should also create a bunch of tests with a "launchd test mock" which pretends to act like launchd and also can do funky things.

File: go/launchd/launchd.go [hunk 1]
```diff
@@ -128,6 +128,28 @@ func (s Service) Restart() error {
...
+	i := 1
+	for time.Now().Sub(t) < wait {
+		status, err := s.LoadStatus()
+		if err != nil {
+			return nil, err
+		}
+		if status != nil {
+			return status, nil
+		}
+		// Tell user we're waiting for status after 4 seconds, every 4 seconds
```
Comment (maxtaco) [inline go/launchd/launchd.go]: `LoadStatus` can return `nil,nil` if control falls off the end of the function, i..e, if `found` isn't `true`.  Is this what happened?

Comment (gabriel) [inline go/launchd/launchd.go]: Yeah... for some reason, very rarely, launchctl doesn't list a service that has been loaded... I think might be a bug in launchd

File: go/launchd/launchd.go [hunk 2]
```diff
@@ -128,6 +128,28 @@ func (s Service) Restart() error {
...
+		if i%4 == 0 {
+			s.info("Waiting for %s to be loaded...", s.label)
+		}
+		time.Sleep(time.Second)
+		i++
+	}
+	return nil, nil
```
Comment (gabriel) [inline go/launchd/launchd.go]: naw.. returning nil means it wasn't found... the caller should check and error in the case if it was indeed an error

Comment (maxtaco) [inline go/launchd/launchd.go]: Then the caller has to check for `nil` and not crash, which I believe isn't the case currently....

Comment (maxtaco) [inline go/launchd/launchd.go]: Ah, I see, it was updated to handle that case cleanly (and just produce an error)

**Comment by maxtaco** [comment]
Well let's be on the lookout for this issue. It might be we're making the problem rarer but we'll still see it (with the 5s timeout) since we're under some race condition we don't fully understand...

**Comment by maxtaco** [comment]
OK to :ship:

**Pull request merged**